x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/IlmHealthIndicatorService.java

```diff
@@ -53,11 +74,72 @@ public class IlmHealthIndicatorService implements HealthIndicatorService {
     );
 
     public static final String AUTOMATION_DISABLED_IMPACT_ID = "automation_disabled";
+    public static final String STAGNATING_INDEX_IMPACT_ID = "stagnating_index";
+    public static final List<HealthIndicatorImpact> AUTOMATION_DISABLED_IMPACT = List.of(
+        new HealthIndicatorImpact(
+            NAME,
+            AUTOMATION_DISABLED_IMPACT_ID,
+            3,
+            "Automatic index lifecycle and data retention management is disabled. The performance and stability of the cluster "
+                + "could be impacted.",
+            List.of(ImpactArea.DEPLOYMENT_MANAGEMENT)
+        )
+    );
+
+    public static final List<HealthIndicatorImpact> STAGNATING_INDEX_IMPACT = List.of(
+        new HealthIndicatorImpact(
+            NAME,
+            STAGNATING_INDEX_IMPACT_ID,
+            3,
+            "Automatic index lifecycle and data retention management cannot make progress on one or more indices. The performance and "
+                + "stability of the indices and/or the cluster could be impacted.",
+            List.of(ImpactArea.DEPLOYMENT_MANAGEMENT)
+        )
+    );
+
+    public static final TimeValue ONE_DAY = TimeValue.timeValueDays(1);
+
+    static final Map<String, RuleConfig> RULES_BY_ACTION_CONFIG = Stream.of(
+        new RuleConfig(RolloverAction.NAME, ONE_DAY, 100, List.of(WaitForActiveShardsStep.NAME)),
```

Comment:
I don't think we want to signal that rollover took one day (I can have a rollover action that just has `max_age: 30d` so no worries there)
For rollover we should only signal if the `WaitForActiveShardsStep` takes one day.

Reply:
that implies a change in this architecture. Will check how to properly do it

Reply:
Yeah ++ I tried to articulate this idea in the previous [comment](https://github.com/elastic/elasticsearch/pull/96092#discussion_r1205317887) hope it makes sense